cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
+                enableEncoding));
+        String expectedDbPrefix = NGSICharsets.encodeMongoDBDatabase(dbPrefix);
+        
+        try {
+            assertEquals(expectedDbPrefix, sink.dbPrefix);
+            System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                    + "-  OK  - 'db_prefix=" + dbPrefix + "' correctly encoded as '" + expectedDbPrefix + "'");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                    + "- FAIL - 'db_prefix=" + dbPrefix + "' wrongly encoded as '" + expectedDbPrefix + "'");
+            throw e;
+        } // try catch // try catch
+    } // testConfigureDBPrefixIsEncodedNewEncoding
+    
+    /**
+     * [NGSIMongoBaseSink.buildCollectionName] -------- When / service-path is notified/defaulted and
+     * data_model=dm-by-service-path, the MongoDB collection name is the concatenation of \<prefix\> and
+     * \<service-path\>.
+     */
+    @Test
+    public void testBuildCollectionNameDMByServicePathRootServicePathOldEncoding() {
+        System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                + "-------- When / service-path is notified/defaulted and data_model=dm-by-service-path, "
+                + "the MongoDB collection name is the concatenation of <prefix> and <service-path>");
+        String collectionPrefix = "sth_";
+        String dbPrefix = "sth_";
+        String dataModel = "dm-by-service-path";
+        String enableEncoding = "false";
+        NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
+        String dbName = "sth_default";
+        String fiwareService = "default";
+        String fiwareServicePath = "/";
+        String entity = "someId=someType";
+        String attribute = "someName";
+        boolean isAggregated = false;
+        String entityId = "someId";
+        String entityType = "someType";
+        
+        try {
+            String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
+                    isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_/";
+            
+            try {
+                assertEquals(expectedCollectionName, collectionName);
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "-  OK  - '" + collectionName + "' was crated as collection name");
+            } catch (AssertionError e) {
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - '" + collectionName + "' was crated as collection name instead of 'sth_/'");
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4